api/v1alpha1/ratelimit_types.go

```diff
@@ -91,6 +92,64 @@ type RateLimitRule struct {
 	// 429 HTTP status code is sent back to the client when
 	// the selected requests have reached the limit.
 	Limit RateLimitValue `json:"limit"`
+	// RequestHitsAddend specifies the number to reduce the rate limit counters
```

Comment:
thanks for introducing this @mathetake !

thinking out loud, and trying to describe user intent 
until now, all we had was "I want to ratelimit at X requests per time unit", now we also have "I want to ratelimit requests based on a total count X, with each response adding to that count by a value (from a response header)"
Im unsure if `cost` is the best word to use here, `hitsAddend` is not intuitive enough imo, another option is `count` e.g. `countPerRequest`

```
costPerRequest: <val>.  // defaults to 1
costPerResonse:             // defaults to 0
  from:
    type: ResponseHeader / Header
    .......
```

cc @envoyproxy/gateway-reviewers @envoyproxy/gateway-maintainers

Reply:
> Im unsure if cost is the best word to use here, hitsAddend is not intuitive enough imo, another option is count e.g. countPerRequest or perRequestCount or simply requestCount

yeah agree not that intuitive, and costPer* sounds good to me

Reply:
btw, our primary usecase is not getting the cost from response header but from a dynamic metadata - so having the explicit from:type:.. won't be much helpful and I would rather expose the opaque "format" which supports both dynamic metadata and headers

Reply:
can we map to format internally, and somehow use a user facing construct instead ?

Reply:
gotcha - will work on it as well as renaming to costPer* stuff tomorrow (in japan 😉)

Reply:
done https://github.com/envoyproxy/gateway/pull/4957/commits/c0694f7c4d846079c037784c67111fcb25048e33